java/memory/src/main/java/io/netty/buffer/NettyArrowBuf.java

```diff
@@ -224,17 +224,8 @@ public ByteBuffer nioBuffer() {
   @Override
   public ByteBuffer nioBuffer(int index, int length) {
     chk(index, length);
-    final ByteBuffer buffer = getDirectBuffer();
-    buffer.position(index).limit(index + length);
-    return buffer;
-  }
-
-  /**
-   * Get this ArrowBuf as a direct {@link ByteBuffer}.
-   * @return ByteBuffer
-   */
-  private ByteBuffer getDirectBuffer() {
-    return PlatformDependent.directBuffer(address, length);
+    UnsafeDirectLittleEndian underlying = arrowBuf.getReferenceManager().getUnderlying();
```

Comment:
this creates another dependency on netty in the ReferenceManager, which we want to avoid.

what's wrong with the getDirectBuffer ?